test/NuGet.Core.Tests/NuGet.Versioning.Test/FloatingRangeTests.cs

```diff
@@ -328,17 +327,822 @@ public void FloatingRange_ToStringMinor()
             Assert.Equal("[1.*, )", range.ToNormalizedString());
         }
 
-        [Fact]
-        public void FloatingRange_FloatMetadata_Invalid()
+        [Theory]
+        [InlineData("1.0.0+*")]
+        [InlineData("1.0.**")]
+        [InlineData("1.*.0")]
+        [InlineData("1.0.*-*bla")]
+        [InlineData("1.0.*-*bla+*")]
+        [InlineData("**")]
+        [InlineData("1.0.0-preview.*+blabla")]
+        [InlineData("1.0.*--")]
+        [InlineData("1.0.*-alpha*+")]
+        [InlineData("1.0.*-")]
+        public void FloatingRange_Parse_Invalid(string floatVersionString)
         {
             // Arrange
             FloatRange range;
 
             // Act
-            var valid = FloatRange.TryParse("1.0.0+*", out range);
+            var valid = FloatRange.TryParse(floatVersionString, out range);
 
             // Assert
             Assert.False(valid);
         }
+
+        [Theory]
+        [InlineData("1.0.0-preview.*")]
+        [InlineData("1.0.*-bla*")]
+        [InlineData("1.0.*-*")]
+        [InlineData("1.0.*-preview.1.*")]
+        [InlineData("1.0.*-preview.1*")]
+        [InlineData("1.0.0--")]
+        [InlineData("1.0.0-bla*")]
+        [InlineData("1.0.*--*")]
+        [InlineData("1.0.0--*")]
+        public void FloatingRange_Parse_Valid(string floatVersionString)
+        {
+            // Arrange
+            FloatRange range;
+
+            // Act
+            var valid = FloatRange.TryParse(floatVersionString, out range);
+
+            // Assert
+            Assert.True(valid);
```

Comment:
Can range verification added in the test as well?

Reply:
Can you clarify what specifically you are referring to?